### ISSUE TITLE
Typescript improvements

### ISSUE THREAD

**Issue by theetrain** [carbon-design-system/carbon-components-svelte#1616]
As advised in https://github.com/carbon-design-system/carbon-components-svelte/discussions/1614#discussioncomment-4618257

- [ ] Switch from `svelte.JSX` to `svelte/elements`
- [ ] Decide whether to use `svelte-package` or augment `sveld`

[`svelte-package`](https://kit.svelte.dev/docs/packaging) allows us to write components with `<script lang=ts>` and ship `carbon-components-svelte` with `.svelte` and `.d.ts` pairs. Sveld is likely still needed to generate documentation.

**Comment by SimpleProgrammingAU** [comment]
I tried to use `<script lang=ts>` for `TextInput` and assigned a type to one of the prop variables.

Is there some more setup that needs to happen before TypeScript can be adopted?

```
5:50:05 pm [vite] Internal server error: C:/code/carbon-components-svelte/src/TextInput/TextInput.svelte:10:20 Unexpected token

- Did you forget to add a TypeScript preprocessor? See https://github.com/sveltejs/vite-plugin-svelte/blob/main/docs/preprocess.md for more information.
  Plugin: vite-plugin-svelte
  File: C:/code/carbon-components-svelte/src/TextInput/TextInput.svelte:10:20
    8 |     * Set to "char" to enable display the character counter or "word" to display the word count.
    9 |     */
   10 |    export let counter: "char" | "word" | undefined = undefined;
                             ^
   11 |
   12 |    /** Set to `true` to disable the input */
```

**Comment by theetrain** [comment]
Using `lang="ts"` is out of scope for the time being. We may eventually use it after closing #1871 since `@sveltejs/package` can convert `.svelte` files containing `lang="ts"` into respective `.svelte` and `.svelte.d.ts` files; allowing users to consume `carbon-components-svelte`, with type safety, and only optionally needing `vitePreprocess`. 

Please use JSDoc for now, as Sveld can build declarations files and documentation for us.
